qcodes/station.py

```diff
@@ -265,6 +265,19 @@ def __getitem__(self, key):
 
     delegate_attr_dicts = ['components']
 
+    def close_all_registered_instruments(self) -> None:
+        """
+        Closes all instruments that are registered to this `Station`
+        object by calling the :meth:`.base.Instrument.close`-method on
+        each one.
+        The instruments will stay registered as a component to the
+        `Station`.
+        """
+        for k, v in self.components.items():
```

Comment:
This ends up changing the dict while iteration over it. I think you need to do a `for k, v in list(self.components.items())` to clone the dict before iterating over it